src/vm/amd64/unixasmmacros.inc

```diff
@@ -52,7 +52,6 @@ C_FUNC(\Name):
 #endif
 C_FUNC(\Name):
         .cfi_startproc
-        .cfi_def_cfa_offset 8
```

Comment:
Interesting. In my live tree, I have the #372 change and still the GDB shows correct stack trace in the first instructions of the CallDescrWorkerInternal until it hits the lea rsp, which breaks it due to the reason that you fixed in the other place.
It actually seems to me that based on what you've said about `the cfa_offset` being 8 after the `.cfi_startproc`, the `.cfi_def_cfa_offset 8` was in fact a nop, since it has defined the offset to the same value.
I think that it is better to do it the way you did it rather than exactly mimicking the C++ compiler.